t/ui/27-plugin_obs_rsync_selenium.t

```diff
@@ -0,0 +1,152 @@
+            );
+        });
+    return $mock;
+}
+
+sub _port { IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => shift) }
+
+my $daemon;
+my $mock            = Mojolicious->new;
+my $server_instance = process sub {
+    $daemon = Mojo::Server::Daemon->new(app => fake_api_server, listen => [$host]);
+    $daemon->run;
+    _exit(0);
+};
+
+sub start_server {
+    $server_instance->set_pipes(0)->start;
+    sleep 1 while !_port($port);
+    return;
+}
+
+sub stop_server {
+    # now kill the worker
+    $server_instance->stop();
+}
+
+$ENV{OPENQA_CONFIG} = my $tempdir = tempdir;
+my $home_template = path(__FILE__)->dirname->dirname->child('data', 'openqa-trigger-from-obs');
+my $home          = "$tempdir/openqa-trigger-from-obs";
+dircopy($home_template, $home);
+my $url = "http://127.0.0.1:$port/public/build/%%PROJECT/_result?package=000product";
+
+$tempdir->child('openqa.ini')->spurt(<<"EOF");
+[global]
+plugins=ObsRsync
+[obs_rsync]
+home=$home
+project_status_url=$url
+EOF
+
+note("Starting fake api server");
+start_server();
+
+my $driver = call_driver(sub { }, {with_gru => 1});
+
+plan skip_all => $OpenQA::SeleniumTest::drivermissing unless $driver;
+
+$driver->find_element_by_class('navbar-brand')->click();
+$driver->find_element_by_link_text('Login')->click();
+
+$driver->get('/admin/obs_rsync/');
+
+is($driver->find_element('tr#folder_Proj1 .project')->get_text(), 'Proj1', 'Proj1 name');
+is($driver->find_element('tr#folder_Proj1 .lastsync')->get_text(), '190703_143010 forget', 'Proj1 last sync');
+ok(index($driver->find_element('tr#folder_Proj1 .dirtystatus')->get_text(), 'dirty') != -1, 'Proj1 dirty status');
+
+# now we call forget_run_last() and refresh_last_run() and check once again corresponding columns
+$driver->find_element('tr#folder_Proj1 .lastsyncforget')->click();
+$driver->get('/admin/obs_rsync/');
+is($driver->find_element('tr#folder_Proj1 .lastsync')->get_text(), 'no data', 'Proj1 last sync forgotten');
+
+wait_for_ajax;
+$driver->find_element('tr#folder_Proj1 .dirtystatusupdate')->click();
+# now wait until gru picks the task up
+my $retries = 5;
+my $dirty_status;
+while ($retries > 0) {
+    $driver->get('/admin/obs_rsync/');
+    $dirty_status = $driver->find_element('tr#folder_Proj1 .dirtystatus')->get_text();
+    last if index($dirty_status, 'dirty') == -1;
+    sleep(1);
+    $retries = $retries - 1;
+}
+
+ok(index($dirty_status, 'dirty') == -1,     'Proj1 dirty status is not dirty anymore');
+ok(index($dirty_status, 'published') != -1, 'Proj1 dirty is published');
+
+wait_for_ajax;
+# click once again and make sure that timestamp on status changed
+$driver->find_element('tr#folder_Proj1 .dirtystatusupdate')->click();
+$retries = 5;
+my $new_dirty_status;
+while ($retries > 0) {
+    $driver->get('/admin/obs_rsync/');
+    $new_dirty_status = $driver->find_element('tr#folder_Proj1 .dirtystatus')->get_text();
+    last if $dirty_status ne $new_dirty_status;
+    sleep(1);
```

Comment:
could it save a bit of time to sleep less, e.g. `sleep .01` in each iteration?

Reply:
That will require 10 times more iterations and will flood output 10 times more, especially on failures.

Reply:
well, all test output should be covered with `Test::Output` so that by default `prove <test>` does not print anything other than the summary report

Reply:
As I said it is annoying to see 10 times bigger log when troubleshooting is needed.